src/main/java/com/school/project/service/impl/LessonServiceimpl.java

```diff
@@ -0,0 +1,65 @@
+package com.school.project.service.impl;
+
+import com.school.project.exception.LessonNotFoundException;
+import com.school.project.model.entities.Lesson;
+import com.school.project.repository.LessonRepository;
+import com.school.project.service.LessonService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class LessonServiceimpl implements LessonService {
+
+    @Autowired
+    LessonRepository lessonRepository;
+
+    @Override
+    public Lesson createLesson(Lesson lesson) {
+        return lessonRepository.save(lesson);
+    }
+
+    @Override
+    public Lesson getLesson(Long id) {
+        Optional<Lesson> byId = lessonRepository.findById(id);
+        if (!byId.isPresent()) throw new LessonNotFoundException("Lesson with id : " + id + " is not found");
+        return byId.get();
+    }
+
+    @Override
+    public List<Lesson> getAllLessonByGroup(Long id) {
```

Comment:
shouldn't it be getAllLessonsByGroup? you are returning a list of items, not one item